examples/next-openai-pages/pages/chat/stream-chat-files-url/index.tsx

```diff
@@ -0,0 +1,113 @@
+import { upload } from '@vercel/blob/client';
+import { Attachment } from '@ai-sdk/ui-utils';
+
+export default function Page() {
+  const { messages, input, handleSubmit, handleInputChange, isLoading } =
+    useChat({
+      api: '/api/stream-chat',
+    });
+
+  const [attachments, setAttachments] = useState<Attachment[]>([]);
+  const [isUploading, setIsUploading] = useState<boolean>(false);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  return (
+    <div className="flex flex-col gap-2">
+      <div className="flex flex-col p-2 gap-2">
+        {messages.map(message => (
+          <div key={message.id} className="flex flex-row gap-2">
+            <div className="w-24 text-zinc-500 flex-shrink-0">{`${message.role}: `}</div>
+
+            <div className="flex flex-col gap-2">
+              {message.content}
+
+              <div className="flex flex-row gap-2">
+                {message.experimental_attachments?.map((attachment, index) => (
+                  <img
+                    key={`${message.id}-${index}`}
+                    className="w-24 rounded-md"
+                    src={attachment.url}
+                    alt={attachment.name}
+                  />
+                ))}
+              </div>
+            </div>
+          </div>
+        ))}
+      </div>
+
+      <form
+        onSubmit={event => {
+          if (isUploading) {
+            alert('Please wait for the files to finish uploading.');
+            return;
+          }
+
+          handleSubmit(event, {
+            experimental_attachments: attachments,
+          });
+
+          setAttachments([]);
+
+          if (fileInputRef.current) {
+            fileInputRef.current.value = '';
+          }
+        }}
+        className="flex flex-col gap-2 fixed bottom-0 p-2 w-full"
+      >
+        <div className="flex flex-row gap-2 fixed right-2 bottom-14 items-end">
+          {Array.from(attachments)
+            .filter(attachment => attachment.mimeType?.startsWith('image/'))
+            .map(attachment => (
+              <div key={attachment.name}>
+                <img
+                  className="w-24 rounded-md"
+                  src={attachment.url}
+                  alt={attachment.name}
+                />
+                <span className="text-sm text-zinc-500">{attachment.name}</span>
+              </div>
+            ))}
+        </div>
+        <input
+          type="file"
+          onChange={async event => {
+            if (event.target.files) {
+              setIsUploading(true);
+
+              for (const file of Array.from(event.target.files)) {
+                const blob = await upload(file.name, file, {
+                  access: 'public',
+                  handleUploadUrl: '/api/file',
+                });
+
+                setAttachments(prevAttachments => [
+                  ...prevAttachments,
+                  {
+                    name: file.name,
+                    mimeType: blob.contentType,
```

Comment:
wondering if `mimeType` or `contentType` is better for our attachment object?

Reply:
We could align and also call it contentType.

Reply:
but we could also leave as is to have consistent examples

Reply:
I agree – updated!